src/Microsoft.AspNetCore.Dispatcher/DispatcherValueEndpointSelector.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Dispatcher
+{
+    public class DispatcherValueEndpointSelector : EndpointSelector
```

Comment:
This is a brute force (and wrong) implementation. I wanted to save the real thing for @jbagga